tests/php/Shortcodes/ImageShortcodeProviderTest.php

```diff
@@ -101,65 +100,79 @@ public function testShorcodeRegenrator()
                 $image->ID,
                 $image->Link()
             ),
-            $parser->parse(sprintf(
-                '[image id="%d" alt="" title=""]',
-                $image->ID
-            )),
+            $parser->parse(
+                sprintf(
+                    '[image id="%d" alt="" title=""]',
+                    $image->ID
+                )
+            ),
             "Shortcode regeneration properly handles empty attributes"
         );
     }
 
     public function testShortcodeHandlerAddsDefaultAttributes()
     {
         $image = $this->objFromFixture(Image::class, 'imageWithoutTitle');
-        $parser = new ShortcodeParser();
-        $parser->register('image', [ImageShortcodeProvider::class, 'handle_shortcode']);
-
-        $this->assertEquals(
-            sprintf(
-                '<img src="%s" alt="">',
-                $image->Link()
-            ),
-            $parser->parse(sprintf(
-                '[image id="%d"]',
-                $image->ID
-            ))
+        $this->assertImageTag(
+            [
+                'src' => $image->Link(),
+                'alt' => '',
+                'width' => '300',
+                'height' => '300',
+                'title' => false,
+                'id' => false,
+            ],
+            sprintf('[image id=%d]', $image->ID),
+            'Shortcode handler adds default attributes'
         );
     }
 
     public function testShortcodeHandlerDoesNotResampleToNonIntegerImagesSizes()
     {
         $image = $this->objFromFixture(Image::class, 'imageWithoutTitle');
-        $parser = new ShortcodeParser();
-        $parser->register('image', [ImageShortcodeProvider::class, 'handle_shortcode']);
-
-        $this->assertEquals(
-            sprintf(
-                '<img src="%s" alt="" width="50%%" height="auto">',
```

Comment:
This confused me for a bit. The way to escape a `%` in `sprintf` is to double it.